src/metabase/driver/common.clj

```diff
@@ -289,8 +288,8 @@
   "Transducer that given a sequence of `values`, returns the most common base type."
   ((comp (filter some?) (take column-info-sample-size) (map class))
    (fn
-     ([] (java.util.HashMap. {:type/* 1}))
-     ([^java.util.HashMap freqs klass]
+     ([] (java.util.HashMap. {nil 0})) ; fallback to keep `max-key` happy if no values
+     ([^java.util.HashMap freqs, klass]
       (.put freqs klass (inc (.getOrDefault freqs klass 0)))
       freqs)
      ([freqs]
```

Comment:
This resolution strategy is lacking. Eg. For numbers it would make more sense to pick the either the first common ancestor type or have some sense of type reification (eg. a mix of ints and floats should be treated as floats no matter the frequencies).

Reply:
can you file an issue for that work and we can get to it in the future?